include/mbedtls/mbedtls_config.h

```diff
@@ -522,6 +522,25 @@
  */
 //#define MBEDTLS_AES_FEWER_TABLES
 
+/**
+ * \def MBEDTLS_AES_ONLY_128_BIT_KEY_LENGTH
+ *
+ * Use only 128-bit keys in AES operations to save ROM.
+ *
+ * Uncommenting this macro removes support for AES operations that are using 192
+ * or 256-bit keys.
+ *
+ * Tradeoff: Uncommenting this macro reduces ROM footprint by ~1116 bytes.
```

Comment:
I don't think we can say "by 1,116 bytes" unconditionally:

Using `size` on an armclang 6.19 build with `--target=arm-arm-none-eabi -mcpu=cortex-m33+nodsp -Oz` I get the following (no AESNI or AESCE)

| orig:ca4ca9a2f |    pr:7451 |      Delta | Filename |
|---------------:|-----------:|-----------:|:---------|
|    7540, 0 |    7320, 0 |     -220+0 | `aes.o` |
|    2930, 0 |    2958, 0 |      +28+0 | `cmac.o` |
|    2078, 0 |    1936, 0 |     -142+0 | `ctr_drbg.o` |
|    6145, 0 |    6193, 0 |      +48+0 | `gcm.o` |
|    2737, 0 |    2799, 0 |      +62+0 | `nist_kw.o` |
|    3654, 0 |    3691, 0 |      +37+0 | `version_features.o` |
|            |            |     -187+0 | TOTAL |

With `MBEDTLS_SELF_TEST`, `MBEDTLS_VERSION_C` and `MBEDTLS_VERSION_FEATURES` disabled I get

| orig:ca4ca9a2f |    pr:7451 |      Delta | Filename |
|---------------:|-----------:|-----------:|:---------|
|    4350, 0 |    4042, 0 |     -308+0 | `aes.o` |
|    1286, 0 |    1272, 0 |      -14+0 | `ctr_drbg.o` |
|            |            |     -322+0 | TOTAL |

Reply:
We definitely can't give a number of bytes for code size, not only because it depends on the compiler, but because it depends on the architecture! Not everyone is compiling for Arm.

We can give a rough ballpark: it reduces the size of the AES code by about 10% (or whatever figure you end up with).

Reply:
> I don't think we can say "by 1,116 bytes" unconditionally

That's correct. I give `1116 bytes` that's because it's measured by `code_size_compare.py` listed in [description](https://github.com/Mbed-TLS/mbedtls/pull/7451#issue-1672799021). But I don't understand why we measure code size with extra options enabled or disabled. What I understand is we measure original code size with `MBEDTLS_AES_ONLY_128_BIT_KEY_LENGTH` disabled then we measure improved code size with `MBEDTLS_AES_ONLY_128_BIT_KEY_LENGTH` enabeld. As the only introduction is `MBEDTLS_AES_ONLY_128_BIT_KEY_LENGTH` in this PR.

I totally agree we can't give an accurate number of bytes for code size saving as it depends on a bunch of options. This `1116` bytes followed [2890](https://github.com/Mbed-TLS/mbedtls/pull/2890/files#diff-f781da22676e488056bc0453f79e0e4dee75eb68433a7ae66e257d17e50cfd89R607) and `MBEDTLS_CAMELLIA_SMALL_MEMORY` in `development`. As both of them give a number of bytes but I can't reproduce this number so I think accuracy of the value might be not crucial.

As @gilles-peskine-arm suggested, write something like `reduces the size of the AES code by about 10%` is a good option. But my concern is percentage might be in a big difference on different architecture and compiler. See code size measurement in [issue description](https://github.com/Mbed-TLS/mbedtls/pull/7451#issue-1672799021). It's `-0.23%` on `x86` with `gcc 8.4.0` but `-0.61%` on `aarch32` with `armclang 6.19`.